calendarsubscription/calendarsubscription.go

```diff
@@ -24,6 +29,36 @@ type CalendarSubscription struct {
 	token string
 }
 
+type iCalOptions struct {
+	Shifts          []oncall.Shift `json:"s,omitempty"`
+	ReminderMinutes []int          `json:"r,omitempty"`
+	Version         string         `json:"v,omitempty"`
+}
+
+var iCalTemplate = template.Must(template.New("ical").Parse(`
+	BEGIN:VCALENDAR
+	PRODID:-//GoAlert//{{.Version}}//EN
+	CALSCALE:GREGORIAN
+	METHOD:PUBLISH
+	{{range .Shifts}}
+	BEGIN:VEVENT
+	SUMMARY:On-Call Shift
+	DTSTART:{{.Start}}
+	DTEND:{{.End}}
+	END:VEVENT
+	{{end}}
+
+	{{range .ReminderMinutes}}
+	BEGIN:VALARM
+	ACTION:DISPLAY
+	DESCRIPTION:REMINDER
+	TRIGGER:-PT{{.}}M
+	END:VALARM
+	{{end}}
+
+	END:VCALENDAR
+`))
```

Comment:
Based on my understanding of the RFC, I think we need to adjust a few things to follow the spec:
- Move `BEGIN:VCALENDAR` to the first line
- Remove the tabs/indentation before each line
- `VALARM` should happen inside the `VEVENT`
- Replace LF with CRLF (e.g. `strings.ReplaceAll(`BEGIN....`, "\n", "\r\n")`)